stream/src/main/java/org/apache/kafka/streaming/processor/internals/RecordQueue.java

```diff
@@ -31,22 +31,24 @@
 
     private final SourceNode source;
     private final TopicPartition partition;
-    private final ArrayDeque<StampedRecord> fifoQueue = new ArrayDeque<>();
-    private final TimestampTracker<ConsumerRecord<Object, Object>> timeTracker = new MinTimestampTracker<>();
+    private final ArrayDeque<StampedRecord> fifoQueue;
+    private final TimestampTracker<ConsumerRecord<Object, Object>> timeTracker;
 
     private long partitionTime = TimestampTracker.NOT_KNOWN;
 
-    /**
-     * Creates a new instance of RecordQueue
-     *
-     * @param partition        partition
-     * @param source           source node
-     */
     public RecordQueue(TopicPartition partition, SourceNode source) {
         this.partition = partition;
         this.source = source;
+
+        this.fifoQueue = new ArrayDeque<>();
```

Comment:
Why did you move these trivial variable initializations to constructor?
Is it the recommended Kafka coding style? Just wondering.